backend/src/index.spec.ts

```diff
@@ -58,105 +75,155 @@ describe('Test apollo server queries', () => {
       { id: 'post1', title: 'Item 1', votes: 0, voters: [], author: {} }
     ]
     const userData = [
-      { name: 'user1', posts: [postData[0]] }
+      { id: 'userid1', name: 'user1', email: 'user1@example.org', password: bcrypt.hashSync('user1password', 10), posts: [postData[0]] }
     ]
     postData[0].author = userData[0]
 
-    const server = new ApolloServer({
-      typeDefs,
-      resolvers,
-      dataSources: () => ({
-        posts: new PostsDataSource(postData, userData)
-      })
-    })
+    const { query } = setupServer(postData, userData)
     const GET_USERS = '{ users { name }}'
-    const { query } = createTestClient(server)
     const res = await query({ query: GET_USERS })
     expect(res.errors).toBeUndefined()
     expect(res.data.users.length).toEqual(1)
     expect(res.data.users[0].name).toEqual('user1')
   })
 
+  it('upvote a post with invalid token returns error', async () => {
+    const postData = [
+      { id: 'post1', title: 'Item 1', votes: 0, voters: [], author: {} }
+    ]
+    const userData = [
+      { id: 'userid1', name: 'user1', email: 'user1@example.org', password: bcrypt.hashSync('user1password', 10), posts: [postData[0]] }
+    ]
+    postData[0].author = userData[0]
+
+    const { mutate } = setupServer(postData, userData, 'invalidToken')
+    const UPVOTE = 'mutation { upvote(id: "post1") { votes }}'
+    const res = await mutate({ mutation: UPVOTE })
+    expect(res.errors.length).toEqual(1)
+    expect(res.errors[0].message).toEqual('Not Authorised!')
+  })
+
+  it('upvote a post with valid token increases count', async () => {
+    const postData = [
+      { id: 'post1', title: 'Item 1', votes: 0, voters: [], author: {} }
+    ]
+    const userData = [
+      { id: 'userid1', name: 'user1', email: 'user1@example.org', password: bcrypt.hashSync('user1password', 10), posts: [postData[0]] }
+    ]
+    postData[0].author = userData[0]
+    let server = setupServer(postData, userData, 'invalidToken')
+    const LOGIN = 'mutation {  login (email: "user1@example.org", password: "user1password") }'
+    const res = await server.mutate({ mutation: LOGIN })
```

Comment:
Your `setupServer` method seems to return the return value of `createTestClient`. So maybe rename your variable from `server` to `client`?